google/cloud/bigtable/value.cc

```diff
@@ -339,18 +339,26 @@ std::ostream& operator<<(std::ostream& os, Value const& v) {
   return StreamHelper(os, v.value_, v.type_, StreamMode::kScalar);
 }
 
+Status MakeDepthExceededError() {
+  return internal::InternalError("Nested value depth exceeds 10 levels",
+                                 GCP_ERROR_INFO());
+}
+
 // NOLINTNEXTLINE(misc-no-recursion)
-Status Value::TypeAndArrayValuesMatch(
-    google::bigtable::v2::Type const& type,
-    google::bigtable::v2::Value const& value) {
+Status TypeAndArrayValuesMatch(google::bigtable::v2::Type const& type,
+                               google::bigtable::v2::Value const& value,
+                               int depth) {
+  if (depth > 10) {
+    return MakeDepthExceededError();
+  }
```

Comment:
Where did the value `10` originate?  It seems arbitrarily low.

In any case, it probably deserves a named constant rather than being repeated in four places.  And, I'd pass the limit to `MakeDepthExceededError()` (or use the named constant there too).